packages/transaction-controller/src/TransactionController.ts

```diff
@@ -944,6 +945,33 @@ export class TransactionController extends BaseController<
     this.update({ transactions: this.trimTransactionsForState(transactions) });
   }
 
+  /**
+   * Add params to transactionMeta.
+   *
+   * @param transactionMetaId - ID of the transaction.
+   * @param securityAlertResponse - New params to be added to tarnsaction meta.
+   */
+  updateSecurityAlertResponse(
+    transactionMetaId: string,
+    securityAlertResponse: SecurityAlertResponse,
+  ) {
+    if (!transactionMetaId || !securityAlertResponse) {
+      return;
+    }
+    const { transactions } = this.state;
+    const index = this.state.transactions.findIndex(
+      ({ id }) => id === transactionMetaId,
+    );
+    if (index >= 0) {
```

Comment:
Rather than allowing missing transactions, should we keep the API consistent and throw as we do in other update methods?

```
if (!transactionMeta) {
  throw new Error(
    `Cannot update security alert response as no transaction metadata found`,
  );
```